test/NuGet.Core.Tests/NuGet.Packaging.Test/RepositoryMetadataTests.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Packaging.Core;
+using Xunit;
+
+namespace NuGet.Packaging.Test
+{
+    public class RepositoryMetadataTests
+    {
+        [Fact]
+        public void RepositoryMetadataDefaultConstructor()
+        {
+            var metadata = new RepositoryMetadata();
+
+            Assert.Equal(string.Empty, metadata.Type);
+            Assert.Equal(string.Empty, metadata.Url);
+            Assert.Equal(string.Empty, metadata.Branch);
+            Assert.Equal(string.Empty, metadata.Commit);
+        }
+
+        [Theory]
+        [MemberData(nameof(EqualsRepositoryMetadataData))]
+        public void RepositoryMetadataEquals(RepositoryMetadata metadata1, RepositoryMetadata metadata2, bool equals)
+        {
+            Assert.Equal(equals, metadata1 == metadata2);
+            Assert.NotEqual(equals, metadata1 != metadata2);
+
+            if (metadata1 != null)
+            {
+                Assert.Equal(equals, metadata1.Equals(metadata2));
+            }
+            if (metadata1 != null && metadata2 != null)
+            {
+                Assert.Equal(equals, metadata1.GetHashCode() == metadata2.GetHashCode());
+            }
+        }
+
+        public static TheoryData EqualsRepositoryMetadataData
+        {
+            get
+            {
+                var template = new RepositoryMetadata("type", "url", "branch", "commit");
+
+                return new TheoryData<RepositoryMetadata, RepositoryMetadata, bool>
+                {
+                    { template, template, true },
+                    { new RepositoryMetadata(), new RepositoryMetadata(), true },
+                    { new RepositoryMetadata("type", "url", "branch", "commit"), template, true },
```

Comment:
Can you add capitalization tests for the branch and commit sha?

Reply:
Done.